sci-wcalc/coupled_stripline_syn.man

.\" Copyright (c), 2005 Dan McMahill
.\" Do not edit this directly.  Edit the XML source file instead
.\"

.TH coupled_stripline_syn "" "" "Wcalc" "Wcalc Commands"
.B coupled_stripline_syn
- Coupled stripline transmission line synthesis

.SH SYNOPSIS

[w_out, s_out, l_out] = 
coupled_stripline_syn(z0, k, z0e, z0o, w, s, h, l, tmet, rho, rough, er, tand, f, flag)


.SH INPUT PARAMETERS

.TP 10
z0
: desired characteristic impedance (Ohms)
.TP 10
k
: desired coupling coefficient
.TP 10
z0e
: desired even mode characteristic impedance (Ohms)
.TP 10
z0o
: desired odd mode characteristic impedance (Ohms)
.TP 10
w
: width of coupled stripline line (meters)
.TP 10
s
: spacing between coupled stripline lines (meters)
.TP 10
h
: substrate thickness (meters)
.TP 10
l
: length of coupled_stripline line (meters)
.TP 10
tmet
: metal thickness (meters)
.TP 10
rho
: resistivity of wire (relative to copper)
.TP 10
rough
: conductor surface roughness (meters-rms)
.TP 10
er
: substrate relative permitivitty
.TP 10
tand
: substrate loss tangent
.TP 10
f
: synthesis frequency (Hz)
.TP 10
flag
: indicates which parameters should be optimized
    
.RS
.TP 10
flag=1
: synthesize based on characteristic impedance and coupling coefficient
.TP 10
flag=0
: synthesize based on even and odd mode characteristic impedances
.RE


.SH OUTPUT VALUES

.TP 10
w_out
: calculated metal width
.TP 10
s_out
: calculated gap
.TP 10
l_out
: calculated metal length
.SH DESCRIPTION

Function to synthesize the physical dimenstions of a
coupled stripline transmission line to achieve a desired characteristic
impedance and electrical length.  The desired characteristic
impedance and some of the physical
dimensions of the coupled_stripline line are given as inputs and the
remaining dimensions are calculated.

The line may be specified either in terms of an even mode and
odd mode characteristic impedance or a characteristic impedance and 
coupling coefficient.

The model accounts for dispersion (frequency dependent propagation
velocity).


.NF



              |<--W-->|<-S->|<--W-->|
               _______       -------
              | metal |     | metal | <- rho,rough
    ----------------------------------------------
   (  dielectric                         /|\     (
    )   er,tand                       H   |       )
   (                                     \|/     (
    ----------------------------------------------
    XXXXXXXXXXXXXXXXXX ground XXXXXXXXXXXXXXXXXXXX



.FI

.SH EXAMPLE
.nf

.TP
 // mils to meters conversion factor
sf=25.4e-6;
.TP
 // desired impedance 
z0=75;
.TP
 // desired coupling
k=0.3;
.TP
 // desired electrical length (degrees)
elen=90;
.TP
 // width (we'll synthesize this)
w=0;
.TP
 // spacing (we'll synthesize this)
s=0;
.TP
 // height
h=15*sf;
.TP
 // length
l=1100*sf;
.TP
 // metal thickness
tmet=1.4*sf;
.TP
 // resitivity of copper
rho=1.72e-8;
.TP
 // metal surface roughness
rough=0.05*sf;
.TP
 // relative dielectric constant
er=4.5;
.TP
 // loss tangent
tand=0.01;
.TP
 // frequency
f=2.4e9;
.TP
 // synthesize from z0 and k (instead
flag=1;
.TP
 // of z0e and z0o)

[w_out,s_out,l_out] = ...
    coupled_stripline_syn(z0,k,elen,w,s,h,l,tmet,rho,rough,er,tand,f,flag);
disp(sprintf('w=%8.4g mils, s=%8.4g mils, l=%8.4g mils',w_out/sf,s_out/sf,l_out/sf));
.fi
.SH SEE ALSO
stdio-wcalc(1),
air_coil_calc(), air_coil_syn(), coax_calc(), coax_syn(), coplanar_calc(), coplanar_syn(), coupled_microstrip_calc(), coupled_microstrip_syn(), coupled_stripline_calc(), coupled_stripline_syn(), ic_microstrip_calc(), ic_microstrip_syn(), microstrip_calc(), microstrip_syn(), stripline_calc(), stripline_syn()
.SH AUTHOR

Dan McMahill

.SH BUGS

None known


NEWS
$Id: NEWS,v 1.10 2009/02/28 03:26:18 dan Exp $

Wcalc -- history of user-visible changes. 

Copyright (c) 1998, 1999, 2000, 2001, 2002, 2003, 2004, 2005
2006, 2007, 2008, 2009 Dan McMahill

Please report bugs on the sourceforge project page.

===============
Version 1.1
===============

Added series/parallel RC equivalent circuit calculator.

Added series/parallel RL equivalent circuit calculator.

Added self and mutual inductance of two rectangular bars.

Added coplanar waveguide model.

Added coupled stripline model.

Corrected the Q calculations for low frequency (not skindepth) region
in air core inductors.  Also corrected Q calculation for conductors
other than copper.

Corrected the calculation of incremental conductance for the
microstrip model.

Corrected a bug in the calculation of conductor losses in the
microstrip model.

Added/enabled code for calculating conductor losses in the coupled
microstrip model.

When compiling with gtk-2.10 or newer, use gtkPrint for printing.
This gives a more professional/standard print dialog and also enables
printing under windows.

Added a preliminary Dutch translation.

Converted the figures to all use solid grayscale fills instead of
pattern fills as the former prints better.

Several updates to the build system to moderize it a bit.  Uses
AM_CONDITIONALS more appropriately, uses AC_PROG_LIBTOOL instead of
AM_PROG_LIBTOOL, etc.

Update the win32 build script to generate an installer that works
correctly under vista.

Add .wc file associations under vista.

Update the octave build infrastructure to work with octave-3.0 and
newer where mex support is built into mkoctfile instead of a
standalone mex shell script.

Switch to GPL licensing.

Create several utility functions to make it faster to build new gtk
forms.  This is only visible to developers.

Improved autogen.sh to check versions of several of the auto* tools to
help make it easier to diagnose any issues which may come up.

Repair compilation with the SunPRO compilers.


===============
Version 1.0
===============
Replaced the alert dialog box in the gtk version with a log
window.  This prevents a problem when many alerts are generated
and the user is presented with many many alert windows.

Added win32 support along with a build script for a windows installer.

Fixed a bug which gave lots of error/warning messages in the cgi
frontend for coupled microstrip.

===============
Version 0.9
===============
First public release.

===============
Version 0.1 
===============
Initial tagged version.  This does good analysis and synthesis
of microstrip.  Considered stable for this purpose.  However,
open/save/print and units other than mils are not supported yet.



octave-wcalc/coupled_stripline_syn.m
%COUPLED_STRIPLINE_SYN - Coupled stripline transmission line synthesis 
%  
%  SYNOPSIS
%  
%  [w_out, s_out, l_out] = coupled_stripline_syn(z0, k, z0e, z0o, w, s, h, l,
%  tmet, rho, rough, er, tand, f, flag)
%  
%  PARAMETERS
%  
%  INPUT PARAMETERS
%  
%    * z0 : desired characteristic impedance (Ohms)
%    * k : desired coupling coefficient
%    * z0e : desired even mode characteristic impedance (Ohms)
%    * z0o : desired odd mode characteristic impedance (Ohms)
%    * w : width of coupled stripline line (meters)
%    * s : spacing between coupled stripline lines (meters)
%    * h : substrate thickness (meters)
%    * l : length of coupled_stripline line (meters)
%    * tmet : metal thickness (meters)
%    * rho : resistivity of wire (relative to copper)
%    * rough : conductor surface roughness (meters-rms)
%    * er : substrate relative permitivitty
%    * tand : substrate loss tangent
%    * f : synthesis frequency (Hz)
%    * flag : indicates which parameters should be optimized
%        + flag=1 : synthesize based on characteristic impedance and coupling
%          coefficient
%        + flag=0 : synthesize based on even and odd mode characteristic
%          impedances
%  
%  OUTPUT VALUES
%  
%    * w_out : calculated metal width
%    * s_out : calculated gap
%    * l_out : calculated metal length
%  
%  DESCRIPTION
%  
%  Function to synthesize the physical dimenstions of a coupled stripline
%  transmission line to achieve a desired characteristic impedance and electrical
%  length. The desired characteristic impedance and some of the physical
%  dimensions of the coupled_stripline line are given as inputs and the remaining
%  dimensions are calculated. The line may be specified either in terms of an even
%  mode and odd mode characteristic impedance or a characteristic impedance and
%  coupling coefficient. The model accounts for dispersion (frequency dependent
%  propagation velocity).
%  
%  
%                |<--W-->|<-S->|<--W-->|
%                 _______       -------
%                | metal |     | metal | <- rho,rough
%      ----------------------------------------------
%     (  dielectric                         /|\     (
%      )   er,tand                       H   |       )
%     (                                     \|/     (
%      ----------------------------------------------
%      XXXXXXXXXXXXXXXXXX ground XXXXXXXXXXXXXXXXXXXX
%  
%  
%  EXAMPLE
%  
%   % mils to meters conversion factor
%  sf=25.4e-6;
%   % desired impedance
%  z0=75;
%   % desired coupling
%  k=0.3;
%   % desired electrical length (degrees)
%  elen=90;
%   % width (we'll synthesize this)
%  w=0;
%   % spacing (we'll synthesize this)
%  s=0;
%   % height
%  h=15*sf;
%   % length
%  l=1100*sf;
%   % metal thickness
%  tmet=1.4*sf;
%   % resitivity of copper
%  rho=1.72e-8;
%   % metal surface roughness
%  rough=0.05*sf;
%   % relative dielectric constant
%  er=4.5;
%   % loss tangent
%  tand=0.01;
%   % frequency
%  f=2.4e9;
%   % synthesize from z0 and k (instead
%  flag=1;
%   % of z0e and z0o)
%  
%  [w_out,s_out,l_out] = ...
%      coupled_stripline_syn(z0,k,elen,w,s,h,l,tmet,rho,rough,er,tand,f,flag);
%  disp(sprintf('w=%8.4g mils, s=%8.4g mils, l=%8.4g mils',w_out/sf,s_out/sf,l_out/sf));
%  
%  SEE ALSO
%  
%  air_coil_calc
%  air_coil_syn
%  coax_calc
%  bars_calc
%  coax_syn
%  coplanar_calc
%  coplanar_syn
%  coupled_microstrip_calc
%  coupled_microstrip_syn
%  coupled_stripline_calc
%  coupled_stripline_syn
%  ic_microstrip_calc
%  ic_microstrip_syn
%  microstrip_calc
%  microstrip_syn
%  parallel_rc_p2s
%  parallel_rc_s2p
%  parallel_rl_p2s
%  parallel_rl_s2p
%  stripline_calc
%  stripline_syn
%  
%  AUTHOR
%  
%  Dan McMahill
%  
%  BUGS
%  
%  None known
%  
%  Please report any bugs on the Wcalc Sourceforge Project Page
%  


% Copyright (c) 2001-2009 Dan McMahill
% All rights reserved.
%
% This program is free software; you can redistribute it and/or modify
% it under the terms of the GNU General Public License as published by
% the Free Software Foundation; version 2 of the License.
% 
% This program is distributed in the hope that it will be useful,
% but WITHOUT ANY WARRANTY; without even the implied warranty of
% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
% GNU General Public License for more details.
% 
% You should have received a copy of the GNU General Public License
% along with this program; if not, write to the Free Software
% Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
%
